spec/controllers/shopping_list_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+require_relative '../support/controller_auth'
+
+RSpec.describe ShoppingListController, type: :controller do
+  extend ControllerAuth
+  describe 'GET #index' do
+    let(:user) { FactoryBot.create(:user) }
+    before { user.save }
+
+    let(:recipe) { FactoryBot.create(:recipe, user:) }
+    before { recipe.save }
+
+    context 'when user is logged in' do
+      it 'should return a 200 response to general_shopping_list route' do
+        get 'index', params: { user_id: user.id, recipe_id: recipe.id }
+        expect(response).to have_http_status(200)
+      end
+
+      it 'should have test: "Shopping List"' do
+        get 'index', params: { user_id: user.id, recipe_id: recipe.id }
+        expect(response.body).to include('Shopping List')
+      end
+    end
+  end
```

Comment:
- [ ] you have some failing tests in your application , kindly fix it 🙏 

![image](https://user-images.githubusercontent.com/89282221/230734946-c0efaa9d-c232-45d6-a9a9-8a8ffb7f6f69.png)